### DISCUSSION TITLE
"every step must define a `uses` or `run` key" I am getting above mentioned error. suggest possible solution

### DISCUSSION THREAD

**Discussion by abulhassanraza** [community/community#26608] [Actions]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/d/2d112388d3d9f5c007700110c4de40eb4499241f.png" data-download-href="/uploads/short-url/6qGcLsi24XVLZRTXi1LeHvtkNpZ.png?dl=1" title="Screenshot (4)" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/75258301/181098663-595945d1-844e-4c01-8ce7-b3241aa82fbc.png" alt="Screenshot (4)" data-base62-sha1="6qGcLsi24XVLZRTXi1LeHvtkNpZ" width="666" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d112388d3d9f5c007700110c4de40eb4499241f_2_666x500.png, https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d112388d3d9f5c007700110c4de40eb4499241f_2_999x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d112388d3d9f5c007700110c4de40eb4499241f_2_1332x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/d/2d112388d3d9f5c007700110c4de40eb4499241f_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screenshot (4)</span><span class="informations">1400×1050 74.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by airtower-luna** [comment]
<p>The second step in your list has only a <code>name</code>, which should probably be part of the next step (which doesn’t have one).</p>

**Comment by abulhassanraza** [comment]
<p>will you explain more specifically. I am unable to get your point. I have a little knowledge of coding.</p>

**Comment by airtower-luna** [comment]
<p>The <code>steps</code> part of the workflow is a list. In YAML each list item is marked by a dash at the same indentation level, like so:</p>
<pre><code class="lang-yaml">steps:
  - name: step 1
    run: this command
  - name: step 2
    run: that command
</code></pre>
<p>Each step (list item) must have a <code>run</code> or <code>uses</code> field to define what the step does.</p>
<p>The second step in your workflow (line 26 in the screenshot) has only a name. In line 27 there’s a third step (new item because it’s starting with a dash) without a name. Its <code>uses: actions/setup-node</code> fits the <code>name: Setup Node.js environment</code> in the previous step, so I assume the <code>uses</code> and <code>with</code> should be part of the second step instead and you just typed a dash where a space should be in line 27.</p>

**Comment by kingthorin** [comment]
<p>Line 26/27 of your screen shot both have hyphens so the <code>uses</code> bit isn’t considered to be “in” the named step.</p>
<p>Edit: also 24 uses the checkout action without being in a named step, I don’t know if that’s a problem though. You’d have to check the docs to see if you can declare a uses element like that. (I don’t know off the top of my head)</p>

**Comment by Simran-B** [comment, marked as answer]
<p>Step names are optional. The log will display "Run " if you only specify <code>uses</code> and “Run &lt;1st line of commands&gt;” (possibly truncated) if you only provide <code>run</code>.</p>

**Comment by abulhassanraza** [comment]
<p>I appreciate your effort for me. I have learned my mistake. May God give you reward</p>
